### PR TITLE
Scenario Update Event (Add/Remove UE)

### PR DESCRIPTION
**Pull request by dilallkx**
**CHANGES:**
- Copyright 2020 update in generated code
- New SCENARIO-UPDATE event to add, remove or modify nodes in the active scenario
  - REST API + backend support only
  - Currently supports only add & remove UE nodes with no child processes
- meep-model methods to add/remove nodes from scenario
- UT updates
- Added Location Service & RNIS to UT execution script
  - Enabled coverage report

**TESTING:**
- Cypress & UT pass
- Manual testing of new SCENARIO-UPDATE event

### CODE REVIEW

File: go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go [hunk 1]
```diff
@@ -707,6 +712,54 @@ func sendEventPoasInRange(event dataModel.Event) (error, int, string) {
...
+	case mod.ScenarioAdd:
+		description = "Added nodes ["
+		for i, node := range event.EventScenarioUpdate.Nodes {
+			if i != 0 {
+				description += ", "
+			}
+			description += node.Name
+		}
+		description += "]"
+		err = sbxCtrl.activeModel.AddScenarioNodes(&event.EventScenarioUpdate.Nodes)
```
Comment (roymx) [inline go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go]: I recommend adding node first, then based on error code, build the description.
e.g. Added nodes[a,b,c] Failed to add [d,e,f]

File: go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go [hunk 2]
```diff
@@ -707,6 +712,54 @@ func sendEventPoasInRange(event dataModel.Event) (error, int, string) {
...
+	case mod.ScenarioRemove:
+		description = "Removed nodes ["
+		for i, node := range event.EventScenarioUpdate.Nodes {
+			if i != 0 {
+				description += ", "
+			}
+			description += node.Name
+		}
+		description += "]"
+		err = sbxCtrl.activeModel.RemoveScenarioNodes(&event.EventScenarioUpdate.Nodes)
```
Comment (roymx) [inline go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go]: I recommend removing node first, then based on error code, build the description.
e.g. Removed nodes[a,b,c] Failed to remove [d,e,f]

File: go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go [hunk 3]
```diff
@@ -707,6 +712,54 @@ func sendEventPoasInRange(event dataModel.Event) (error, int, string) {
...
+		err := errors.New("Malformed request: missing EventScenarioUpdate")
+		return err, http.StatusBadRequest, ""
+	}
+	if len(event.EventScenarioUpdate.Nodes) == 0 {
+		err := errors.New("Malformed request: missing Nodes")
+		return err, http.StatusBadRequest, ""
+	}
+
+	// Perform necessary action on scenario
+	switch event.EventScenarioUpdate.Action {
```
Comment (roymx) [inline go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go:730]: I believe that current implementation only supports adding Physical locations of type UE
Logic to ensure that only UEs nodes are added is missing

File: go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go [hunk 4]
```diff
@@ -66,10 +66,13 @@ const fieldSandboxName = "sandbox-name"
...
-const eventTypeMobility = "MOBILITY"
-const eventTypeNetCharUpdate = "NETWORK-CHARACTERISTICS-UPDATE"
-const eventTypePoasInRange = "POAS-IN-RANGE"
-const eventTypeOther = "OTHER"
+const (
+	eventTypeMobility       = "MOBILITY"
+	eventTypeNetCharUpdate  = "NETWORK-CHARACTERISTICS-UPDATE"
+	eventTypePoasInRange    = "POAS-IN-RANGE"
+	eventTypeScenarioUpdate = "SCENARIO-UPDATE"
+	eventTypeOther          = "OTHER"
```
Comment (roymx) [inline go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go:74]: "OTHER" event type should be removed

File: go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go (hunk 1 above)
Comment (dilallkx) [reply go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go]: Fixed in latest commit.
NOTE: Current implementation stops adding nodes if a node fails to be added successfully.

File: go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go (hunk 2 above)
Comment (dilallkx) [reply go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go]: Fixed in latest commit.
NOTE: Current implementation stops removing nodes if a node fails to be removed successfully.

File: go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go (hunk 4 above)
Comment (dilallkx) [reply go-apps/meep-sandbox-ctrl/server/sandbox-ctrl.go:74]: Removed.

**Pull request merged**